vmm/src/vmm_config/logger.rs

```diff
@@ -1,12 +1,53 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+//! Auxiliary module for configuring the logger.
 extern crate serde_json;
 
-use std::fmt::{Display, Formatter, Result};
+use libc::O_NONBLOCK;
+use std::fmt::{Display, Formatter};
+use std::fs::{File, OpenOptions};
+use std::io::{LineWriter, Write};
+use std::os::unix::fs::OpenOptionsExt;
+use std::path::PathBuf;
 
 use self::serde_json::Value;
 
+type Result<T> = std::result::Result<T, std::io::Error>;
+
+/// Structure `LoggerWriter` used for writing to a FIFO.
+pub struct LoggerWriter {
+    line_writer: LineWriter<File>,
+}
+
+impl LoggerWriter {
+    /// Create and open a FIFO for writing to it.
+    pub fn new(fifo_path: &str) -> Result<LoggerWriter> {
+        let fifo = PathBuf::from(fifo_path);
+        match OpenOptions::new()
+            .custom_flags(O_NONBLOCK)
+            .read(true)
+            .write(true)
+            .open(&fifo)
```

Comment:
Nit: let t = OpenOptions:: ...... open()?; Would save you an indention level.

Reply:
Indeeed, take a look.